src/Dapr.Actors/Runtime/TimerInfo.cs

```diff
@@ -0,0 +1,136 @@
+﻿// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr.Actors.Runtime
+{
+    using System;
+    using System.Globalization;
+    using System.IO;
+    using System.Text;
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Dapr.Actors.Resources;
+
+    internal class TimerInfo
+    {
+        private readonly TimeSpan minTimePeriod = Timeout.InfiniteTimeSpan;
+
+        public TimerInfo(
+            string callback,
+            byte[] state,
+            TimeSpan dueTime,
+            TimeSpan period)
+        {
+            this.ValidateDueTime("DueTime", dueTime);
+            this.ValidatePeriod("Period", period);
+            this.Callback = callback;
+            this.Data = state;
+            this.DueTime = dueTime;
+            this.Period = period;
+        }
+
+        public string Callback { get; private set; }
+
+        public TimeSpan DueTime { get; private set; }
+
+        public TimeSpan Period { get; private set; }
+
+        public byte[] Data { get; private set; }
+
+        internal static async Task<TimerInfo> DeserializeAsync(Stream stream)
```

Comment:
I'd suggest turning this functionality into a JsonConverter, and putting the `JsonConverterAttribute` on the `TimerInfo` class. You can keep all the same functionality, but it's composable.